pkg/backend/httpstate/state.go

```diff
@@ -37,44 +37,18 @@ import (
 	"github.com/pulumi/pulumi/pkg/v3/secrets"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/config"
-	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
 )
 
-// cloudUpdate is an implementation of engine.Update backed by remote state and a local program.
-type cloudUpdate struct {
-	context context.Context
-	backend *cloudBackend
-
-	update      client.UpdateIdentifier
-	tokenSource *tokenSource
-
-	root   string
-	proj   *workspace.Project
-	target *deploy.Target
-}
-
-func (u *cloudUpdate) GetRoot() string {
-	return u.root
-}
-
-func (u *cloudUpdate) GetProject() *workspace.Project {
-	return u.proj
-}
-
-func (u *cloudUpdate) GetTarget() *deploy.Target {
-	return u.target
-}
-
-func (u *cloudUpdate) Complete(status apitype.UpdateStatus) error {
```

Comment:
This method now lives at `(b *cloudBackend) completeUpdate(...)` -- see later on in the diff